File: src/main/kotlin/predcompiler/compilation/evaluation/statevariables/IdentityMapping.java
package predcompiler.compilation.evaluation.statevariables;

import java.util.HashMap;

public class IdentityMapping extends AbstractStateToRobustnessMapping {

	/**
	 * Variable that will be transformed into a Column within a real valuation set.
	 */
	private String targetVariable;

	public IdentityMapping(String targetVariable) {
		this.targetVariable = targetVariable;
	}

	@Override
	public float mapRobustness(HashMap<String, Float> state) {
		return state.get(targetVariable);
	}

	@Override
	public String getPredicateRepresentation() {
		return targetVariable;
	}

	@Override
	public String[] getPossibleParams() {
		return new String[0];
	}

} // IdentityMapping

File: src/main/kotlin/predcompiler/compilation/evaluation/statevariables/IdentityMapping.kt
package predcompiler.compilation.evaluation.statevariables

class IdentityMapping(
    /**
     * Variable that will be transformed into a Column within a real valuation set.
     */
    private val targetVariable: String,
) : AbstractStateToRobustnessMapping() {
    override fun mapRobustness(state: HashMap<String, Double>): Double {
        return state[targetVariable] ?: return -1.0
    }

    override val predicateRepresentation: String
        get() {
            return targetVariable
        }
} // IdentityMapping

